shared/src/telemetry/posthog.rs

```diff
@@ -42,11 +44,21 @@ pub fn init(event_name: &str) {
             return;
         }
 
+        // For privacy reasons, we generate an anonymous UUID for each new deployment
+        let uuid_file = format!("/var/lib/postgresql/data/{}_uuid", extension_name);
+        let distinct_id = if Path::new(&uuid_file).exists() {
+            fs::read_to_string(uuid_file).unwrap_or_else(|_| uuid::Uuid::new_v4().to_string())
```

Comment:
You might want to perform some validation here to make sure that what we're reading from the file is actually a UUID. I don't really see a security risk, but we're otherwise sending unsanitized input data in our posthog request.

Reply:
Looks like you can just use `Uuid::try_parse(uuid_string)`.

Reply:
I used `Uuid::parse_Str` instead, could you please take a look?